src/stores/quiz/actions.ts

```diff
@@ -3,20 +3,33 @@ import type { QuizAnswers } from '@/domain/QuizAnswer'
 import { post } from '@/services/http'
 import { quizAnswer } from './state'
 import type { Role } from '@/domain/Role'
+import type { Answer } from '@/domain/Answer'
+import type { QuizResult } from '@/domain/QuizResult'
+import { ErrorMessage } from '@/services/messages'
 
-const initializeQuizAnswers = (knowledges: Knowledge[], quizAnswers: QuizAnswers) => {
-  quizAnswers.answers = knowledges.map((knowledge) => ({
-    knowledgeId: Number(knowledge.id),
+const initializeQuizAnswers = (knowledges: Knowledge[]) => {
+  quizAnswer.value.answers = knowledges.map((knowledge) => ({
+    knowledgeId: knowledge.id,
     answer: 0
-  }))
+  })) as Answer[]
 }
 
-async function getQuizResult(quizAnswer: QuizAnswers) {
-  return await post('/calculate-score', quizAnswer)
+async function getQuizResult(quizAnswer: QuizAnswers): Promise<QuizResult> {
+  try {
+    const response = await post<QuizResult>('/api/quiz/evaluate-answers', quizAnswer)
+    return response
+  } catch (error) {
+    ErrorMessage('Failed to fetch quiz result:')
+    throw error
+  }
 }
 
 async function setSelectedRole(role: Role) {
-  quizAnswer.value.roleId = role.id!
+  if (role.id !== undefined) {
```

Comment:
You don't need this if here. It should not be possible to pass undefined to this function